concepts/&str/about.md

```diff
@@ -1,3 +1,69 @@
 # About
 
-TODO
+There are two ways that Rust implements an array of characters: [str][str] and [String][String].
```

Comment:
```suggestion
There are two ways that Rust implements an array of characters: [`str`][str] and [`String`][String].
```

It seems like we are establishing a convention of using code formatting when discussing types. Sound fair?

Reply:
I had it that way at first, but then noticed that it hid the link unless you mouse over it, so I removed the back ticks. But I don't feel strongly about it if you want me to put them back in.

Reply:
I prefer to render types with code formatting. If exercism's stylesheet hides links in code formatting, that's a stylesheet bug; we should fix the stylesheet instead of leaving out semantic markers to work around it.